### ISSUE TITLE
I can help you for your testing

### ISSUE THREAD

**Issue by arsiesis** [codmpm/node-red-contrib-loxone#9]
Hello,
I am a teacher in belgium
We have loxone material in school.
Your work could allow us to make our lessons more interactive by connecting loxone via nodered to a home simulator

I would like to help you ...
I saw that you had difficulty doing your testing on hardware.
I can put at your disposal, a PC with teamviewer and a loxone server connect on the same network. you can then turn on (PC and loxone server) whenever you want via a simple HTTP request. and do all your test as you like ... I can put this in place this Wednesday, 4 October ... if you are interested, just tell me how to send you in PM, the connection information.

Thank you again for your work

**Comment by codmpm** [comment]
Hi arsiesis,

this is great to hear. Thank you for the compliments :-)

And thank you specially for the offer for access to a testing environment. Currently the people at the loxforum try to help me out. The biggest problem is to have a "real world" installation and see what people develop with the possible connection from node-red to loxone.

I would be glad if your students will build things and test everything they can imagine. In my opinion the problems they run into will tell us more about the use cases than testing some made up ones ;-)

So, I politely decline your offer as I've got a miniserver for testing purposes. But I am very keen on what you all will build and hope you can give me an update from time to time 👍 

See my Blog (https://allgeek.de, german) for some examples. You can contact me via blog@allgeek.de

Cheers,
Patrik

**Comment by keesnobel** [comment]
hello patrik,

First of all, good work.
but unfortunately since last update from loxone to vs 9, your loxone node does not want to reconnect after a disconnected connection.
As soon as I reboot the loxone mini server, the node will no longer connect and it says connection closed.

Kees

**Comment by codmpm** [comment]
Thank you. Sadly my funded miniserver is not here yet and I can't test.

Have you ticked the "activate connection" checkbox in the config node? And what error message shows up in the node-red-log?

What version of node-red-contrib-loxone are you running?

**Comment by keesnobel** [comment]
I have tried all encryption options and activated the "activation connection" checkbox.

The node-red-contrib-loxone version is 0.6.0, node -v is v6.11.3, nodered is v0.17.5
All this on a raspberry pi zero w

And on another server, node -v is v8.6.0 (also the same problem as soon as I reboot the loxone mini server)
 
If the rpi's Wi-Fi connection is lost, do not re-connect the node
When I press a button (gpio pulldown input), this is the output of control out,

> TypeError: Can not read property 'send_control_command' of null.


and after a while...

> Miniserver connection error: undefined


Too bad that the miniser has not arrived yet, thanks for your work

**Comment by mabunixda** [comment]
May be you can add a ping command to test if the (w)lan connection is available all the time when the problem occures.

I am running already a unstable version right now of v9 and had no problems starting up with node-red-contrib-loxone - but i am running the docker container version of node-red ( node 6.11.4 ) and everything is cable connected.

**Comment by codmpm** [comment]
I agree. Wi-Fi is not the best solution for timing based applications. Alternatively try to connect an usb-ethernet adapter via USB-OTG to the zero and check if the problem still occurs.

Nevertheless, I will test what happens when the miniserver reboots and when the ethernet connection is dropped.

**Comment by mabunixda** [comment]
I can see this error only when i deploy within node-red ... 

@codmpm I can monitor the logs - i gonna do an upgrade by tomorrow ;)

**Comment by codmpm** [comment]
I think I have a missing `hasOwnProperty` somewhere, when the connection is not present. But I have to wait till I get a miniserver again.

**Comment by mabunixda** [comment]
I did the upgrade now ... attached you find a [loxone-upgrade.log](https://github.com/codmpm/node-red-contrib-loxone/files/1400034/loxone-upgrade.log)

On line 32 - 54 and 75-97 are new deployments within node-red.
After 106 the miniserver is upgrading and afterwards rebooting.

I had to restart the node-red container to bring it up working again. 
After starting the flows on startup there is the same "Cannot read property" error because the connection to miniserver is connected afterwards. The timestamps are on seconds so it is hard to analyse if this should happen or if this is just a bad timing problem...

> node-red_1      | 19 Oct 23:23:54 - [info] [loxone-miniserver:9a604121.d4647] Miniserver connected (172.16.0.33:80) using Token-Enc
> node-red_1      | 19 Oct 23:23:55 - [info] [loxone-miniserver:9a604121.d4647] got structure file 2017-10-19 23:16:58
> node-red_1      | 19 Oct 23:25:54 - [info] [loxone-miniserver:9a604121.d4647] keepalive (2ms)
